packages/runtime/test/integration/pool-transaction-fee-test.ts

```diff
@@ -0,0 +1,90 @@
+import { types } from "@algo-builder/web";
+import { ExecParams } from "@algo-builder/web/build/types";
+import { assert } from "chai";
+
+import { RUNTIME_ERRORS } from "../../src/errors/errors-list";
+import { AccountStore, Runtime } from "../../src/index";
+import { ALGORAND_ACCOUNT_MIN_BALANCE } from "../../src/lib/constants";
+import { useFixture } from "../helpers/integration";
+import { expectRuntimeError } from "../helpers/runtime-errors";
+
+describe("Pooled Transaction Fees Test", function () {
+  useFixture("app-update");
+  const minBalance = ALGORAND_ACCOUNT_MIN_BALANCE; // 1000 to cover fee
+  let john = new AccountStore(1e30);
+  let alice = new AccountStore(minBalance);
+  let bob = new AccountStore(minBalance);
+
+  let runtime: Runtime;
+
+  this.beforeEach(async function () {
+    runtime = new Runtime([john, alice, bob]); // setup test
+  });
+
+  // helper function
+  function syncAccounts (): void {
+    john = runtime.getAccount(john.address);
+    bob = runtime.getAccount(bob.address);
+    alice = runtime.getAccount(alice.address);
+  }
+
+  it("Should pass if second account doesn't pay fees and first account is covering fees", () => {
+    const amount = 1e4 + 122;
+    const initialBalance = john.balance();
+    // group with fee distribution
+    const groupTx: ExecParams[] = [
+      {
+        type: types.TransactionType.TransferAlgo,
+        sign: types.SignType.SecretKey,
+        fromAccount: john.account,
+        toAccountAddr: alice.address,
+        amountMicroAlgos: amount,
+        payFlags: { totalFee: 2000, flatFee: true }
```

Comment:
So in this case `flatFee` should be set as true? If we don't set it, then transaction fee won't be pooled?

Reply:
no it is not necessary, updated it.